httpbin/middleware.go

```diff
@@ -17,13 +41,6 @@ func cors(h http.Handler) http.Handler {
 		respHeader.Set("Access-Control-Allow-Origin", origin)
 		respHeader.Set("Access-Control-Allow-Credentials", "true")
 
-		if r.Method == "OPTIONS" {
```

Comment:
Sorry, but it's not obvious to me why this code needed to be moved into a different middleware invoked before this `cors` middleware.  Would you mind explaining?

Reply:
I moved it in another handler just for clarity and because I couldn't figure out how to name a function that handles both CORS and `HEAD`/`OPTIONS` request methods :slightly_smiling_face:  There's no technical reason they can't be in the same handler, but the same could probably be said for `logger()` and `limitRequestSize()`. 

Also, due to the way these handlers are composed, the execution order is `cors`, `optionsAndHead`, `logger`,  `limitRequestSize` and finally the `mux`. But now that I think about it, it shuld probably be `cors`, `logger`, `optionsAndHead`,  `limitRequestSize`, otherwise the log will treat `HEAD` requests as `GET`.